### DISCUSSION TITLE
React Aria ListBox link support

### DISCUSSION THREAD

**Discussion by TrevorRice** [adobe/react-spectrum#5242] [General]
Hi, all!

I was just curious — what was the rationale for [`ListBox`](https://react-spectrum.adobe.com/react-aria/ListBox.html#links) (and by extension `Select` and `Combobox`) to support links? And when would this be recommended over a set of [`Link`](https://react-spectrum.adobe.com/react-aria/Link.html) components? I've always been under the assumption that listboxes should _not_ be used for navigation.

From Adrian Roselli's [Be Careful Using 'Menu'](https://adrianroselli.com/2023/05/be-careful-using-menu.html#Listbox):

> An ARIA listbox is also a form control, not a navigation tool and not an application menu.

From [APG's Listbox pattern](https://www.w3.org/WAI/ARIA/apg/patterns/listbox/):

> [...] if an option contains a semantic element, such as a heading, screen reader users will not have access to the semantics. In addition, the interaction model conveyed by the listbox role to assistive technologies does not support interacting with elements inside of an option. Because of these traits of the listbox widget, it does not provide an accessible way to present a list of interactive elements, such as links, buttons, or checkboxes.

By attaching `role="option"` to an `<a>` tag, its implicit `link` role is overridden, providing no information to a screen reader user, for instance, that what they're about to interact with will navigate them somewhere as demonstrated in Safari and VoiceOver below.

| `ListBox` (`<a href="" role="option">`) | `<a href="#>` |
| --- | --- |
| _Google, text, (3 of 4)_ | _link, Google_  |
| ![CleanShot 2023-10-12 at 07 17 25@2x](https://github.com/adobe/react-spectrum/assets/3644075/db298f86-f56d-4b24-a01f-ba00d39328a3) | ![CleanShot 2023-10-12 at 07 17 02@2x](https://github.com/adobe/react-spectrum/assets/3644075/2df805c1-bf29-476a-a931-e2cb426e1be8) |

**Comment by devongovett** [comment]
For ListBox in particular, the main use case is not when used standalone but when used inside a ComboBox/autocomplete. For example, our documentation search uses links on its items to perform navigation, just like the Google autocomplete. Unfortunately, there's no way to have an element take two different roles (option and link), and a ComboBox is expected to have a ListBox attached which only supports options. We discussed potentially adding an additional aria-label or aria-roledescription to indicate that an item is a link but punted for the first release. Sometimes links are also just implementation details of the application/framework and not that relevant to the end user, so it is debatable whether it should _always_ do that or only sometimes. Perhaps we will make improvements as we get more use cases and feedback.

**Reply by TrevorRice** [reply]
I appreciate the added context!
